### PR TITLE
feat: add email, sms, and metamask extension

### LINKED ISSUE #133: Enable metamask snap in cosmos-kit
Which one?

https://docs.cosmology.zone/cosmos-kit/integrating-wallets/leap-metamask-comos-snap
https://docs.cosmology.zone/cosmos-kit/integrating-wallets/cosmos-metamask-extension

### LINKED ISSUE #134: Enable web3auth magic link (email)
https://web3auth.io/docs/auth-provider-setup/social-providers/email-passwordless

### PR DESCRIPTION
**Pull request by chalabi2**
fixes #133
fixes #134

## Summary by CodeRabbit

- **New Features**
	- Introduced conditional logo rendering for the 'Cosmos MetaMask Extension' across multiple components.
	- Added new components for email and SMS input functionality.
	- Enhanced wallet integration with new login methods (Email and SMS).
	- Expanded wallet filtering options to include 'Email' and 'SMS'.
	- Implemented a fee estimation feature in the send form.
	- Updated styling for button hover states in dark mode across various components.

- **Bug Fixes**
	- Improved error handling for wallet connections and rendering logic.

- **Chores**
	- Updated dependencies in `package.json` to ensure compatibility and access to new features.

- **Documentation**
	- Added new exports for `EmailInput` and `SMSInput` components in the index file.

### CODE REVIEW

**Review by fmorency — CHANGES_REQUESTED** [review]
Email, SMS and Metamask doesn't work.

Email & SMS



Metamask

**Comment by chalabi2** [comment]
@fmorency this will only work if you pull the branch from the PR on your cosmos kit repo. Please dont merge until we get cosmos kit to update web3auth. I have tested it all locally and it works great.

**Comment by fmorency** [comment]
I tested it and it works well. The SMS input has some rendering problems here.

**Comment by fmorency** [comment]
The latest `@cosmis-kit/web3auth` package doesn't contain a `cjs` folder and this is breaking my build.

File: components/react/views/SMSInput.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+      <Formik
+        initialValues={{ phoneNumber: '' }}
+        validationSchema={validationSchema}
+        onSubmit={values => {
+          const fullNumber = `${selectedCountryCode}-${values.phoneNumber}`;
+          if (phoneRegExp.test(fullNumber.replace('-', ''))) {
+            onSubmit(fullNumber);
+          }
+        }}
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply components/react/views/SMSInput.tsx:76]: this is required as per the web3auth docs

**Review by fmorency — APPROVED** [review]

**Pull request merged**